posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py

```diff
@@ -2199,29 +2229,103 @@ async def test_insert_into_s3_activity_when_using_distributed_events_recent_tabl
         file_format=file_format,
         batch_export_schema=None,
         batch_export_model=model,
+        is_backfill=is_backfill,
+        backfill_details=BackfillDetails(
+            backfill_id=str(uuid.uuid4()),
+            start_at=backfill_start_at,
+            end_at=data_interval_end,
+            is_earliest_backfill=False,
+        )
+        if is_backfill
+        else None,
     )
 
-    with override_settings(
-        BATCH_EXPORT_S3_UPLOAD_CHUNK_SIZE_BYTES=5 * 1024**2,
-        BATCH_EXPORT_DISTRIBUTED_EVENTS_RECENT_ROLLOUT=1 if use_distributed_events_recent_table else 0,
-    ):  # 5MB, the minimum for Multipart uploads
-        records_exported = await activity_environment.run(insert_into_s3_activity, insert_inputs)
+    class MockClickHouseClient:
+        """Helper class to mock ClickHouse client."""
+
+        def __init__(self):
+            self.mock_client = mock.AsyncMock(spec=ClickHouseClient)
+            self.mock_client_cm = mock.AsyncMock()
+            self.mock_client_cm.__aenter__.return_value = self.mock_client
+            self.mock_client_cm.__aexit__.return_value = None
+
+            # Set up the mock to return our async iterator
+            self.mock_client.astream_query_as_arrow.return_value = self._create_record_batch_iterator()
+
+        def expect_query_string(self, expected_query_string: str) -> None:
+            """Assert that the executed query contains the expected string."""
+            assert self.mock_client.astream_query_as_arrow.call_count == 1
+            call_args = self.mock_client.astream_query_as_arrow.call_args
+            query = call_args[0][0]  # First positional argument of the first call
+            assert expected_query_string in query
```

Comment:
Doesn't look like we are calling this method anywhere?

Reply:
yes, you're right, I forgot to delete this in a previous refactor. Will clean up 👍